dask/dataframe/tests/test_dataframe.py

```diff
@@ -907,6 +908,9 @@ def test_map_partitions():
     assert_eq(d.map_partitions(lambda df: df, meta=d), full)
     assert_eq(d.map_partitions(lambda df: df), full)
     result = d.map_partitions(lambda df: df.sum(axis=1))
+    layer = hlg_layer(result.dask, "lambda-")
+    assert not layer.is_materialized(), layer
+    result.dask.validate()
```

Comment:
Minor comment, but `result.dask.validate()` should be called in the `assert_eq` check in the next line, so I don't think we need the extra check here too 

```suggestion
```

Reply:
Thanks, didn't realize that!